src/module/orbbec.cpp

```diff
@@ -1427,9 +1462,34 @@ vsdk::ProtoStruct Orbbec::do_command(const vsdk::ProtoStruct& command) {
                     return device_control::getCameraParams<ob::Pipeline, ob::VideoStreamProfile>(dev->pipe);
                 } else if (key == "create_module_config") {
                     return device_control::createModuleConfig<ViamOBDevice, ob::VideoStreamProfile>(dev);
+                } else if (key == "call_get_properties") {
+                    call_get_properties = true;
                 }
             }
         }  // unlock devices_by_serial_mu_
+        if (call_get_properties) {
+            VIAM_SDK_LOG(info) << "[do_command] calling get_properties";
+            auto const props = get_properties();
+            VIAM_SDK_LOG(info) << "[do_command] get_properties called";
+            vsdk::ProtoStruct resp;
+            resp["supports_pcd"] = props.supports_pcd;
+            resp["intrinsic_parameters"] = vsdk::ProtoStruct{{"width_px", props.intrinsic_parameters.width_px},
+                                                             {"height_px", props.intrinsic_parameters.height_px},
+                                                             {"focal_x_px", props.intrinsic_parameters.focal_x_px},
+                                                             {"focal_y_px", props.intrinsic_parameters.focal_y_px},
+                                                             {"center_x_px", props.intrinsic_parameters.center_x_px},
+                                                             {"center_y_px", props.intrinsic_parameters.center_y_px}};
+            vsdk::ProtoStruct distortion_params;
+            distortion_params["model"] = props.distortion_parameters.model;
+            std::vector<viam::sdk::ProtoValue> proto_params;
+            for (const auto& val : props.distortion_parameters.parameters) {
+                proto_params.emplace_back(val);  // Each double becomes a ProtoValue
+            }
+            distortion_params["parameters"] = viam::sdk::ProtoValue(proto_params);
+            resp["distortion_parameters"] = distortion_params;
+            VIAM_SDK_LOG(info) << "[do_command] get_properties returning";
+            return resp;
+        }
```

Comment:
I'm confused. Why do we need this as a docommand route if get_properties exists as a method?